chapter3/src/3_2/3_2_1/main_3_2_1_2.rs

```diff
@@ -0,0 +1,20 @@
+/**
+ *
+ * chapter 3_2_1_2
+ * io.read
+ *
+ */
+use std::io::Read;
+
+pub fn main() -> std::io::Result<()> {
+    let mut r: &[u8] = b"abcdefg";
+    let mut buffer = [0; 4];
+    match r.read_exact(&mut buffer) {
```

Comment:
`panic!` は対応する必要があるかなと思っています。`?` で伝播してもほぼ同じ結果を得られますし、あるいは `expect` を使用する手があります。

下記のように expect でもう少しエラー内容を書くか、そもそも本書内でもあまりエラーメッセージについては細かく書かれていない印象なので、`?` で済ませるでもよいと思います。

```suggestion
    let result = r.read_exact(&mut buffer).expect("想定できるエラー内容の原因をより詳しく書くとか");
```

Reply:
了解です。
ひとまず、`?`でまとめたcommit作りました